### ISSUE TITLE
stage2: can't import root in a package

### ISSUE THREAD

**Issue by leecannon** [ziglang/zig#12274]
### Zig Version

0.10.0-dev.3362+e863292fe

## main.zig:
```zig
const other = @import("other");

comptime {
    _ = other;
}

pub fn main() void {}
```
## other.zig
```zig
const root = @import("root");

comptime {
    _ = root;
}
```

```
zig2 build-exe main.zig --pkg-begin other other.zig --pkg-end
```

### Expected Behavior

Compiles without error.

### Actual Behavior

```
$ zig2 build-exe main.zig --pkg-begin other other.zig --pkg-end
thread 392769 panic: reached unreachable code
Analyzing other.zig: other.zig:root
    > %2 = import("root") token_offset:1:22 to :1:28
      %3 = break_inline(%1, %2)
    For full context, use the command
      zig ast-check -t other.zig

  in other.zig: other.zig:comptime_0
    > %5 = decl_val("root") token_offset:4:9 to :4:13

/home/lee/src/zig/src/Sema.zig:9895:17: 0x357a4ce in Sema.zirImport (zig)
                unreachable;
                ^
/home/lee/src/zig/src/Sema.zig:755:64: 0x33d7176 in Sema.analyzeBodyInner (zig)
            .import                       => try sema.zirImport(block, inst),
                                                               ^
/home/lee/src/zig/src/Sema.zig:619:45: 0x33d00dc in Sema.analyzeBodyBreak (zig)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/home/lee/src/zig/src/Module.zig:4394:50: 0x31eac4f in Module.semaDecl (zig)
    const result_ref = (try sema.analyzeBodyBreak(&block_scope, body)).?.operand;
                                                 ^
/home/lee/src/zig/src/Module.zig:4035:38: 0x31c7760 in Module.ensureDeclAnalyzed (zig)
    const type_changed = mod.semaDecl(decl_index) catch |err| switch (err) {
                                     ^
/home/lee/src/zig/src/Sema.zig:24034:32: 0x37e1461 in Sema.ensureDeclAnalyzed (zig)
    sema.mod.ensureDeclAnalyzed(decl_index) catch |err| {
                               ^
/home/lee/src/zig/src/Sema.zig:24076:32: 0x35faa44 in Sema.analyzeDeclRef (zig)
    try sema.ensureDeclAnalyzed(decl_index);
                               ^
/home/lee/src/zig/src/Sema.zig:24023:45: 0x37bda8e in Sema.analyzeDeclVal (zig)
    const decl_ref = try sema.analyzeDeclRef(decl_index);
                                            ^
/home/lee/src/zig/src/Sema.zig:5243:31: 0x3571a88 in Sema.zirDeclVal (zig)
    return sema.analyzeDeclVal(block, src, decl);
                              ^
/home/lee/src/zig/src/Sema.zig:727:65: 0x33d5b02 in Sema.analyzeBodyInner (zig)
            .decl_val                     => try sema.zirDeclVal(block, inst),
                                                                ^
/home/lee/src/zig/src/Sema.zig:619:45: 0x33d00dc in Sema.analyzeBodyBreak (zig)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/home/lee/src/zig/src/Module.zig:4394:50: 0x31eac4f in Module.semaDecl (zig)
    const result_ref = (try sema.analyzeBodyBreak(&block_scope, body)).?.operand;
                                                 ^
/home/lee/src/zig/src/Module.zig:4035:38: 0x31c7760 in Module.ensureDeclAnalyzed (zig)
    const type_changed = mod.semaDecl(decl_index) catch |err| switch (err) {
                                     ^
/home/lee/src/zig/src/Compilation.zig:3050:38: 0x2ef6dfe in Compilation.processOneJob (zig)
            module.ensureDeclAnalyzed(decl_index) catch |err| switch (err) {
                                     ^
/home/lee/src/zig/src/Compilation.zig:2911:30: 0x2ee07fd in Compilation.performAllTheWork (zig)
            try processOneJob(comp, work_item);
                             ^
/home/lee/src/zig/src/Compilation.zig:2251:31: 0x2ed8cd3 in Compilation.update (zig)
    try comp.performAllTheWork(main_progress_node);
                              ^
/home/lee/src/zig/src/main.zig:3305:20: 0x2e43e5f in updateModule (zig)
    try comp.update();
                   ^
/home/lee/src/zig/src/main.zig:2994:17: 0x2e0be65 in buildOutputType (zig)
    updateModule(gpa, comp, hook) catch |err| switch (err) {
                ^
/home/lee/src/zig/src/main.zig:219:31: 0x2defb37 in mainArgs (zig)
        return buildOutputType(gpa, arena, args, .{ .build = .Exe });
                              ^
/home/lee/src/zig/src/main.zig:174:20: 0x2deee04 in main (zig)
    return mainArgs(gpa, arena, args);
                   ^
/home/lee/src/zig/lib/std/start.zig:576:37: 0x2df16cf in std.start.main (zig)
            const result = root.main() catch |err| {
                                    ^
[1]    392769 IOT instruction (core dumped)  zig2 build-exe main.zig --pkg-begin other other.zig --pkg-end
```

**Comment by Vexu** [comment]
Will be fixed by #11867

**Comment by andrewrk** [comment]
e6b3eae490e4d0c427034c3530be4ed5a74e2253
